diff --git a/gli/core/save_dds.inl b/gli/core/save_dds.inl
index 62ab305..8f9b551 100644
--- a/gli/core/save_dds.inl
+++ b/gli/core/save_dds.inl
@@ -102,7 +102,7 @@ namespace detail
 		if(Texture.faces() > 1)
 		{
 			GLI_ASSERT(Texture.faces() == 6);
-			Header.CubemapFlags |= detail::DDSCAPS2_CUBEMAP_ALLFACES | detail::DDSCAPS2_CUBEMAP;
+			Header.CubemapFlags |= static_cast<std::uint32_t>(detail::DDSCAPS2_CUBEMAP_ALLFACES) | static_cast<std::uint32_t>(detail::DDSCAPS2_CUBEMAP);
 		}
 
 		// Texture3D
