diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd8f5153..353f7ef1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -586,18 +586,6 @@ ELSE()
   INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
 ENDIF()
 
-IF( NOT IOS )
-  IF( NOT ASSIMP_BUILD_MINIZIP )
-    use_pkgconfig(UNZIP minizip)
-  ENDIF()
-ELSE ()
-  IF( NOT BUILD_SHARED_LIBS )
-    IF( NOT ASSIMP_BUILD_MINIZIP )
-      USE_PKGCONFIG(UNZIP minizip)
-    ENDIF()
-  ENDIF ()
-ENDIF ()
-
 IF ( ASSIMP_NO_EXPORT )
   ADD_DEFINITIONS( -DASSIMP_BUILD_NO_EXPORT)
   MESSAGE( STATUS "Build an import-only version of Assimp." )
